### PR TITLE
Operator linalg_syevd: Symmetric eigendecomposition

### PR DESCRIPTION
**Pull request by mseeger**
There are some issues with float32. I'd like to see what the test suite is doing.
If problems persist, I will deactivate the new operator for float32.

@asmushetzel

### CODE REVIEW

**Comment by mseeger** [comment]
OK, just like on my laptop, the tests fail, due to an extremely weird behaviour of LAPACK ssyevd (float32), which does not happen for dsyevd (float64). We need more time to get to the bottom of this. For now, I will change the code to support float64 only for the new linalg_syevd operator, and update the PR shortly.

**Comment by mseeger** [comment]
OK, I deactivated float32 for the new operator. Now, tests should run through.

**Comment by mseeger** [comment]
@asmushetzel 

Do you understand why this turned into a "review for pull request", and why the test suite does not seem to get started? Just AppVeyor ran. ???

**Comment by asmushetzel** [comment]
Sometimes Jenkins chokes. You may have to push again in order to trigger it. 
I think it is ok. to disable float32 for now. I can try to debug the 32-bit issue when working on Cuda-support.

**Comment by mseeger** [comment]
The build fails for "Python 2: GPU Win". This is not my code (see below).

What should I do? I will try to pull in the latest master and run the tests again.

======================================================================

FAIL: test_operator_gpu.test_batchnorm_training

----------------------------------------------------------------------

Traceback (most recent call last):

  File "C:\Anaconda3\envs\py2\lib\site-packages\nose\case.py", line 197, in runTest

    self.test(*self.arg)

  File "C:\Anaconda3\envs\py2\lib\site-packages\nose\util.py", line 620, in newfunc

    return func(*arg, **kw)

  File "c:\jenkins_slave\workspace\ut-python-gpu\tests\python\gpu\../unittest\test_operator.py", line 969, in test_batchnorm_training

    check_batchnorm_training(stype)

  File "c:\jenkins_slave\workspace\ut-python-gpu\tests\python\gpu\../unittest\test_operator.py", line 909, in check_batchnorm_training

    check_numeric_gradient(test, in_location, mean_std, numeric_eps=1e-2, rtol=0.16, atol=1e-4)

  File "c:\jenkins_slave\workspace\ut-python-gpu\pkg_vc14_gpu\python\mxnet\test_utils.py", line 879, in check_numeric_gradient

    ("NUMERICAL_%s"%name, "BACKWARD_%s"%name))

  File "c:\jenkins_slave\workspace\ut-python-gpu\pkg_vc14_gpu\python\mxnet\test_utils.py", line 466, in assert_almost_equal

    raise AssertionError(msg)

AssertionError: 

Items are not equal:

Error 1.113524 exceeds tolerance rtol=0.160000, atol=0.000100.  Location of maximum error:(1, 0, 1, 0), a=0.000691, b=0.000977

 NUMERICAL_data: array([[[[-2.35421062, -3.84590626],

         [-2.39110589,  4.12830114]],

...

 BACKWARD_data: array([[[[-2.35458016, -3.84661579],

         [-2.39165258,  4.12931108]],

...


----------------------------------------------------------------------

**Comment by mseeger** [comment]
OK, now Python3: MKLML-GPU fails for another unrelated reason:

libdc1394 error: Failed to initialize libdc1394

File: src/operator/tensor/la_op.cc [hunk 1]
```diff
@@ -550,5 +550,79 @@ NNVM_REGISTER_OP(_backward_linalg_gelqf)
 .set_attr<nnvm::TIsBackward>("TIsBackward", true)
 .set_attr<FCompute>("FCompute<cpu>", LaOpBackward<cpu, 2, 2, 4, 1, gelqf_backward>);
 
+NNVM_REGISTER_OP(_linalg_syevd)
+.add_alias("linalg_syevd")
```
Comment (piiswrong) [inline src/operator/tensor/la_op.cc]: linalg_* alias was added for previous ops for backward compatibility. New ops don't need this alias.
_linalg_syevd enables access from mx.sym.linalg.syevd

**Comment by piiswrong** [comment]
@asmushetzel The test failures are irrelevant. This LGTM after removing alias. Do you think it's ready to merge?

File: src/operator/tensor/la_op.cc (hunk 1 above)
Comment (mseeger) [reply src/operator/tensor/la_op.cc]: Great to see we can now use the linalg namespace.

Does this work for all operators, or just for the new ones?
Is it already reflected in the documentation?

**Comment by mseeger** [comment]
OK, removed the alias

**Comment by eric-haibin-lin** [comment]
Could you sync with the master branch? The lint test should pass now

**Comment by asmushetzel** [comment]
Matthias, can you do one more testrun to verify that the lint problem is indeed gone? IMO, this should be ok to merge then.

**Comment by mseeger** [comment]
OK, just pushed the branch, rebased to current master

**Comment by asmushetzel** [comment]
There is one remaining problem. When the alias linalg_seyvd gets removed, also the unit-tests have to be updated to use the linalg.seyvd notion. Currently they still try to use linalg_seyvd which does no longer exist, so they all fail.

**Comment by mseeger** [comment]
OK, I fixed this. Now, all unit tests use linalg.XXX instead of linalg_XXX.

**Comment by asmushetzel** [comment]
MKML-GPU tests are failing because of an unrelated problem with batchnorm-tests. 
Perl-GPU failing because of an unrelated general setup problem. 
Everything else clean. 

IMO, this can be merged now.

**Pull request merged**